angr/state_plugins/unicorn_engine.py

```diff
@@ -1054,6 +1086,19 @@ def setup(self):
         if self.gdt is not None:
             _UC_NATIVE.activate_page(self._uc_state, self.gdt.addr, bytes(0x1000), None)
 
+        if self.state.mode == "tracing":
```

Comment:
I would really prefer we added state options for behavior like this instead of gating it behind the state mode. I'd really like to move away from behaviors that are dependent on the state's mode.

Reply:
Adding a state option to control whether concrete open fd bytes should be passed to native interface makes sense and I agree with that. However, I think the tracing mode check should also be there. Otherwise, we might end up concretizing an fd with unconstrained symbolic bytes which might not be desirable given the amount of solving that would be done.